anton_gadzikovskiy/graph_generator.cpp

```diff
@@ -107,46 +206,168 @@ string print_graph(const Graph& graph) {
   graph_output += "]\n}\n";
   return graph_output;
 }
+
 }  // namespace graph_printing
 
-Graph generate_graph() {
-  auto graph = Graph();
-
-  for (int i = 0; i < kVerticesCount; i++) {
-    graph.add_vertex();
-  }
-
-  graph.add_edge(0, 1);
-  graph.add_edge(0, 2);
-  graph.add_edge(0, 3);
-  graph.add_edge(1, 4);
-  graph.add_edge(1, 5);
-  graph.add_edge(1, 6);
-  graph.add_edge(2, 7);
-  graph.add_edge(2, 8);
-  graph.add_edge(3, 9);
-  graph.add_edge(4, 10);
-  graph.add_edge(5, 10);
-  graph.add_edge(6, 10);
-  graph.add_edge(7, 11);
-  graph.add_edge(8, 11);
-  graph.add_edge(9, 12);
-  graph.add_edge(10, 13);
-  graph.add_edge(11, 13);
-  graph.add_edge(12, 13);
-
-  return graph;
+bool is_generated(double prob) {
+  srand(time(0));
+  return (rand() % 100) <= prob;
+}
+
+const VertexId& get_random_vertex_id(const vector<VertexId>& vertices) {
+  srand(time(0));
+  const int random_id = rand() % vertices.size();
+  return vertices[random_id];
+}
+
+void generate_gray_edges(Graph& graph, int depth, int new_vertices_num) {
+  graph.add_vertex();
+  VertexId new_child_vertex_id = 1;
+  for (int i = 0; i < depth; i++) {
+    double step = 100 / depth;
+    double prob = 100 - step * i;
+    if (i < graph.get_vertices_on_depth().size()) {
+      vector<VertexId> temp_vertices_on_depth =
+          graph.get_vertices_on_depth().at(i);
```

Comment:
1) Лучше для этого добавить удобный интерфейс
1) Зачем тебе `temp` префикс?
1) Это не `vertices`, а `vertex_ids`
1) Почему не `const`
1) Почему делаешь копию? Добавь комментарий, если это оправдано
```suggestion
      const auto vertex_ids_on_depth = graph.get_vertex_ids_on_depth(current_depth);
```

Reply:
Исправил название, а копию делал потому что, у меня возникла ошибка, в которой почему-то id новой вершины менялся на какое-то большое число, и создание копии исправило ее.

Reply:
Это потому, что ты изменял объект, по которому итерировался. Нельзя одновременно итерироваться по коллекции и менять её, тогда обращения к памяти указывают на неверные адреса.
Добавь комментарий прямо в коде

Reply:
Добавил